sql/parser/eval.go

```diff
@@ -1179,35 +1180,36 @@ func ParseDate(s DString) (DDate, error) {
 }
 
 // ParseTimestamp parses the timestamp.
-func ParseTimestamp(s DString) (DTimestamp, error) {
+func (ctx EvalContext) ParseTimestamp(s DString) (DTimestamp, error) {
 	str := string(s)
 	t, err := time.Parse(dateFormat, str)
 	if err == nil {
 		return DTimestamp{Time: t}, nil
 	}
-	t, err = time.Parse(timestampFormat, str)
+	t, err = time.Parse(TimestampWithOffsetZoneFormat, str)
 	if err == nil {
 		t = t.UTC()
 		return DTimestamp{Time: t}, nil
 	}
-	t, err = time.Parse(TimestampWithOffsetZoneFormat, str)
+	location := ctx.Location
+	if location == nil {
+		location = time.UTC
+	}
+	t, err = time.ParseInLocation(timestampFormat, str, location)
 	if err == nil {
 		t = t.UTC()
 		return DTimestamp{Time: t}, nil
 	}
-	t, err = time.Parse(timestampWithNamedZoneFormat, str)
-	if err == nil {
-		// Parsing using a named time zone is imperfect for two reasons:
-		// 1. Some named time zones are ambiguous (PST can be US PST and
-		// phillipines PST), and 2. The code needs to have access to the entire
-		// database of named timed zones in order to get some time offset,
-		// and it's not clear what are the memory requirements for that.
-		// TODO(vivek): Implement SET TIME ZONE to set a time zone and use
-		// time.ParseInLocation()
-		return DummyTimestamp, util.Errorf("TODO(vivek): named time zone input not supported")
+	// We do not trust Named Zone parsing without a specified location.
```

Comment:
Why does it look like a typo? We have a series of `time.Parse` calls followed by `if err == nil { return t }`, and this is just the same pattern. The named format is least likely to be used and so it makes sense to put it last.